### ISSUE TITLE
BPF+LLVM limitation: can't pass a C struct by value to another translation unit

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#1803]
This sample BPF program:
```c
#include <solana_sdk.h>
struct foo {};
void foo(struct foo foo); // foo() defined in another translation unit

extern bool entrypoint(const uint8_t *input) {
  struct foo f;
  foo(f);
  return true;
}
```
fails to compile with llvm-7.0:
```
/usr/local/opt/llvm/bin/llc -march=bpf -filetype=obj  -o out/noop.o out/noop.bc
error: <unknown>:0:0: in function entrypoint i32 (i8*): pass by value not supported 0x7fdee0059e08: i64 = GlobalAddress<void (%struct.foo*)* @foo> 0
```
Seems like the function foo from another source file is unable to accept the struct foo by value.

**Comment by jackcmay** [comment]
```rust
fail(CLI.DL, DAG, "pass by value not supported ", Callee);
```
http://llvm.org/doxygen/BPFISelLowering_8cpp_source.html

**Comment by jackcmay** [comment]
Fixed in #2100
